src/components/TradeAssetSearch/hooks/useFetchAndUpsertCustomTokenMarketData.tsx

```diff
@@ -0,0 +1,37 @@
+import { useCallback } from 'react'
+import { useDispatch } from 'react-redux'
+import { marketData as marketDataSlice } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectIsCustomAsset } from 'state/slices/selectors'
+import { store } from 'state/store'
+
+import { getTokenMarketData } from './useGetCustomTokenPriceQuery'
+
+// Custom hook to fetch and upsert market data for custom tokens
+export const useFetchAndUpsertCustomTokenMarketData = () => {
```

Comment:
preferably blocking: This would benefit from wrapping in react-query so cache prevents multiple duplicate calls to zerion, similar to `useGetCustomTokenPricesQuery`. Basically instead of this hook returning a callback, export `queryKey` and `queryFn` from this file, and then in `SearchTermAssetList` you can manually invoke the query via `queryClient.fetchQuery({ .. })`.

Reply:
do we have preference for using react-query for a slice that is build around rtk query setup? I had the same thought, but initially thought of adding to the `marketApi` vs the react-query setup.

Reply:
I've removed this file entirely in https://github.com/shapeshift/web/pull/7370/commits/dedae61a38016ecb062e2d0350b53e20254d0f21 and will re-asses with a Market Data service type approach in a follow-up 🙏